tools/chainsync-loadtest/src/concurrency/ctx.rs

```diff
@@ -0,0 +1,183 @@
+use crate::concurrency::Once;
+use std::fmt;
+use std::ops;
+use std::sync::Arc;
+use tokio::time;
+
+// CtxErr represents an error that Context can return.
```

Comment:
FYI: I didn't spend much time reviewing these libraries - as currently they will be used only by this tool.

But if in future, we want to move them to the 'core' - then we should do another thorough review + tests.

Reply:
yes, this is just a PoC. I'd like to get some traction on the ideas encoded here and some opinions of our Rust experts before we go any further with that. Then we should definitely invest into adding tests and finding a decent location for this code.

Reply:
I'd ask some experts to review the libraries :) @matklad @nagisa could you take a look if you have time